Dockerfiles

kerimovkhikmet opened the pull request:
Had added Dockerfiles for app and db containers. Started on working with docker-compose.

faxx1080 committed:
Fix docs for setting profiles

Add using -Dspring-boot.run.jvmArguments to the notes for how to set up using MySQL.

piyush-garg committed:
Upgrade to wro4j 1.9.0

See gh-650

kerimovkhikmet committed:
created Dockerfiles

kerimovkhikmet committed:
added .env to .gitignore

kerimovkhikmet committed:
test Dockerfile

kerimovkhikmet committed:
testing Dockerfile

kerimovkhikmet committed:
testing Dockerfile

kerimovkhikmet committed:
roll back with git stage

kerimovkhikmet committed:
testing without git clone

kerimovkhikmet committed:
edited copy path

kerimovkhikmet committed:
changed copy path

kerimovkhikmet committed:
changed workdir

kerimovkhikmet committed:
edited copy source folder

kerimovkhikmet committed:
added .m2 directory for cache

kerimovkhikmet committed:
added app user

kerimovkhikmet committed:
using experimental syntax

kerimovkhikmet committed:
adding user

kerimovkhikmet committed:
added docker-compose

kerimovkhikmet committed:
changed docker-compose

kerimovkhikmet committed:
changed pom file

kerimovkhikmet committed:
optimized app dockerfile

kerimovkhikmet committed:
changed layer for a stage

Rudya93 reviewed (changes requested):
Please add comments to files

kerimovkhikmet committed:
added readme and comments

kerimovkhikmet committed:
Update readme.md

kerimovkhikmet committed:
Update readme.md

kerimovkhikmet committed:
edited main readme.md

kerimovkhikmet commented:
> Please add comments

kerimovkhikmet committed:
edited application Dockerfile

kerimovkhikmet committed:
edited docker-compose

kerimovkhikmet committed:
adding swarm

kerimovkhikmet committed:
small fix

kerimovkhikmet committed:
finished project

kerimovkhikmet committed:
edited readme

kerimovkhikmet committed:
fixed readme

CesarionPshebytski committed:
Updated user for mysql

Updated creation of user for mysql db. Previously lack of quotes resulted in creation of incorrect user (petclinic@% instead of just petclinic)

snicoll committed:
Upgrade to Spring Boot 2.3.3

snicoll committed:
Upgrade to spring javaformat 0.0.25

snicoll committed:
Upgrade to Checkstyle 8.32

snicoll committed:
Restore version to 2.3.0.BUILD-SNAPSHOT

akashsolanki committed:
Remove redundant junits dependencies

See gh-670

snicoll committed:
Merge pull request #670 from akashsolanki

* pr/670:
  Remove redundant junits dependencies

Closes gh-670

snicoll committed:
Merge pull request #650 from piyush-garg

* pr/650:
  Upgrade to wro4j 1.9.0

Closes gh-650

snicoll committed:
Upgrade to Spring Boot 2.3.5

Commit:
Upgrade to Spring Boot 2.4.0

Arka-Bandyopadhyay committed:
remove .vscode folder from project repository

Arka-Bandyopadhyay committed:
Downgrade to wro4j 1.8.0

This commit downgrades the wro4j plugin to 1.8.0 as the latest version
requires Java 11.

See gh-707

snicoll committed:
Remove IDE specific configuration files

Closes gh-706

snicoll committed:
Merge pull request #707 from Arka-Bandyopadhyay

* pr/707:
  Downgrade to wro4j 1.8.0
  remove .vscode folder from project repository

Closes gh-707

snicoll committed:
Upgrade to Spring Boot 2.4.1

dlresende committed:
Do not fail packaging if repo info is absent

As part of a process to build a BOSH release, I encountered the error
below while running `./mvnw package`:

```
[ERROR] Failed to execute goal pl.project13.maven:git-commit-id-plugin:3.0.1:revision (default) on project spring-petclinic: Could not complete Mojo execution...: Error: Could not get HEAD Ref, are you sure you have set the dotGitDirectory property of this plugin to a valid path? -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

That's because we are just interested on the source files for the BOSH
release, we don't care about repo metadata (so we exclude them).

Going through the [git-commit-id-maven-plugin
docs](https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/master/maven/docs/using-the-plugin.md)
we learned it is possible to prevent the failure above via
configuration.

We propose then to set
`<failOnUnableToExtractRepoInfo>false</failOnUnableToExtractRepoInfo>`,
enabling use cases similar to ours.

FWIW `failOnNoGitDirectory` is already set to false.

Commit:
Update to Boot 2.4.2

Commit:
Add docker push to CI

Commit:
Only publish docker image from main branch

testforstephen committed:
Configure an inline m2e lifecycle mapping for wro4j-maven-plugin

Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

Commit:
Update Spring Boot to 2.4.5

Commit:
Add javaformat plugin to Eclipse workaround

Commit:
Only attempt to login to docker on main

Commit:
Only attempt to login to docker on main

Commit:
Make build work with Gradle

Commit:
Clarify that a JDK is needed

Commit:
Add note about autocrlf on windows

lsap committed:
Update petclinic_db_setup_mysql.txt

Hey Team, thanks in advance!

DanielFran committed:
Upgrade to Maven 3.8.2

See gh-805

snicoll committed:
Merge pull request #805 from DanielFran

* pr/805:
  Upgrade to Maven 3.8.2

Closes gh-805

snicoll committed:
Upgrade to Spring Boot 2.5.4

simrin051 committed:
Add pagination for owners and vets lists in HTML

Fixes #91

Commit:
Add delimiter to list of pets

Commit:
Tidy up test

arey committed:
Add GitHub actions maven-build.yml

arey committed:
Replace Travis badge by GitHub actions one

Commit:
Ensure real database is used if mysql profile active

trisberg committed:
Add license file

arey committed:
Use the provided maven wrapper

Commit:
Tidy imports

Commit:
Update to Boot 2.5.5

Commit:
Add note on Java version

Commit:
Remove travis config file

Commit:
Update to Spring Boot 2.5.6

Commit:
Convert to SCSS and Bootstrap 5

Commit:
Fix page links

Commit:
No need for jquery

Commit:
Remove redundant interface

Commit:
Alternative approach with libsass-maven-plugin

Commit:
Put CSS generation in a Maven profile

Commit:
Make buttons look normal again

Commit:
Remove unused css files

Commit:
Tidy up properties in build config

Commit:
Fix ignores

nroi committed:
Update docker-compose.yml to version 2.2

The current docker-compose.yml is still in version 1, which is
deprecated: https://docs.docker.com/compose/compose-file/compose-versioning/

So if you try to run `docker-compose up` with a recent version of
docker-compose, the following error occurs:
    (root) Additional property mysql is not allowed

Version 2.2 was chosen because it's not deprecated, but it also has been
released some time ago: Compose version 2.2 requires docker engine
version 1.13.0, which was released on 2017-01-18. So, with this version,
hopefully no users will have any issues either with deprecation warnings
or with a version that is not yet supported by their docker-engine.

Commit:
Use "date" type for date inputs

In 2021 we really should be using the browser's own date picker.

nroi committed:
Add support for PostgreSQL

See #636

Commit:
Update for recent changes in mvn build

Commit:
Upgrade to Spring Boot 2.6.1 and fix Maven build

Commit:
Mention Gradle build in readme

trisberg committed:
Update to Boot 2.6.2

Commit:
Avoid dropping Postgres schema

The postgres user needs to be an owner of the public schema or a
superuser in order to execute the schema.sql for database
initialization.

To avoid that, two changes are required:
 - use `CREATE TABLE IF NOT EXISTS` in order to idempotently create the
   necessary tables
 - handle conflicts when inserting duplicate data

This follows the pattern in use by the
src/main/resources/db/mysql/{data,schema.sql} files.

Authored-by: Shaan Sapra <shsapra@vmware.com>

Commit:
Remove PetRepository and use Owner as aggregate

Owner is really the aggregate root in DDD terms and there is no
need to directly access the Pet entity.

Commit:
No need for exclusion in recent versions

Commit:
Also remove VisitRepository

Rely on Hibernate to do lazy loading on visits.

Commit:
Move Visit to owner package (it's aggregate root)

odrotbohm committed:
Additional tweaks in aggregate model.

Introduced Owner.addVisit(…) to avoid that state transition to live in controller code.

Slightly polished some assertions in ClinicServiceTests.

Commit:
Remove manual id management in child entities

This is reverting a workaround for a Hibernate "feature". There's
no need for the child entities (Pet and Visit) to know about their
parent (foreign key). Hibernate can manage that just fine with a
@JoinColumn. But it needs a nullable foreign key column in the
DB schema. That's the downside. The upside is much less code in
Java.

Commit:
Add actuators to Gradle build

Commit:
Fix lazy loading issue in owner details

Commit:
Alternative approach with eager fetch

Commit:
Add section on building a container

snicoll committed:
Upgrade to Spring Boot 2.6.3

snicoll committed:
Upgrade to nohttp 0.0.10

snicoll committed:
Upgrade to spring javaformat 0.0.31

snicoll committed:
Upgrade to Maven Checkstyle plugin 3.1.2

kerimovkhikmet committed:
Update base image tags and 'docker/**/readme.md' files

Commit:
Remove unused node.version

Commit:
Update jacoco to work with Java 17

noumanurrahman committed:
Fixed Typo In PetClinicIntegrationTests

win777 committed:
Update readme.md

Remove Java 8

win777 committed:
Update readme.md

win777 committed:
Update readme.md

Commit:
Use open session in view and populate model attrs

Open session in view was switched off accidentally a while ago. Also
the mapping changes recently meant that the changes to @Valid model
attributes were not being propagated correctly.

Fixes #946 and #947

Commit:
Make postgres data.sql properly idempotent

Fixes #936

Commit:
Update gradle build boot version

Commit:
Update to Spring Boot 2.7.0

Commit:
Update name of deprecated config property

dsyer committed:
Add devcontainer and gitpod

Also works in codespaces.

dsyer committed:
Relocate deprecated annotation

Commit:
Whitespace tidy

kvaithin committed:
985 remove lastName method param from OwnerController

amaembo committed:
Update javadoc: it incorrectly said to return 'true'

Commit:
Move vscode settings up a level

jdubois committed:
use Java 17 in devcontainers

jdubois committed:
remove unused import

jdubois committed:
Add the Pivotal extension pack to devcontainers (#994)

Add Spring Boot extensions

Commit:
Upgrade to Boot 2.7.1

robertson-neil committed:
Upgrade build.grade to spring-boot-starter 2.7.1 (#1016)

@pivotal-cla This is an Obvious Fix
This fixes a compilation error in PetClinicIntegrationTests
Upgrade to same version as used in pom.xml

shwetatyagi22 committed:
Removed unnecessary semicolons (#998)

Commit:
Mention Gradle at top level in README

Commit:
Upgrade to Gradle 7.5.1 and to Spring Boot 2.7.3

Commit:
Ensure gradle home is set in container

Commit:
Attempt to fix codespaces

martinlippert committed:
launch java tooling in standard mode right away

Commit:
Add support and CI tests for Java 8

Commit:
Change CI step names

ckyeon committed:
Readability improvements & Use stream instread of forEach (#1055)

* Use stream instead of forEach on find PetType logic of parse()

* Improve controllers readability

* Rollback stream instead of for-each

Commit:
Tweak devcontainer so java and permissions work

kerimovkhikmet committed:
Merge pull request #1 from spring-projects/main

Sync with main project

Commit:
Change mvn command in app Dockerfile
